docs/DeveloperGuide.adoc

```diff
@@ -385,6 +385,225 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+
+=== Shortcuts for the user in filling fields
+
+==== Current Implementation
+Due to the nature of the application, the user will input certain information repetitively.
+For example, `role` can only be one of `student` or `tutor`.
+
+Thus, we have created shortcuts to help improving efficiency in filling the fields.
+We currently support shortcuts for `subject`, `level`, `status`, and `role`.
+
+[width="59%",cols="<30%,<35%,<35%",options="header",]
+|=======================================================================
+|Applicable field |Shortcut |Equivalence in full
+|subject | `sub/chem`| `sub/chemistry`
+|subject | `sub/phy`| `sub/physics`
+|subject | `sub/chi`| `sub/chinese`
+|subject | `sub/eng`| `sub/english`
+|level | `lvl/us`| `lvl/upper sec`
+|level | `lvl/ls`| `lvl/lower sec`
+|level | `lvl/up`| `lvl/upper pri`
+|level | `lvl/lp`| `lvl/lower pri`
+|status | `stat/nm`| `stat/not matched`
+|status | `stat/m`| `stat/matched`
+|role | `r/t`| `r/tutor`
+|role | `r/s`| `r/student`
+|=======================================================================
+
+==== Model Implementation
+The shortcut is first validated, then the field information is converted to its full form.
+Below is the function that performs conversion in the Role class. Other classes follow similar implementation.
+[source,java]
+----
+ public String convertToFullRole(String original) {
+      String cur = original.toLowerCase();
+       if (cur == null) {
+           return "";
+        } else if (cur.equals("s")) {
+            cur = "student";
+        } else if (cur.equals("t")) {
+            cur = "tutor";
+        }
+       return cur;
+ }
+----
+[NOTE]
+The user can input shortcuts in case-insensitive fashion.
+For instance, the user may input `add n/jOHN dOE p/98765432 e/johND@eXaMpLe.com a/NUS #04-02 $/50 sub/PHy lvl/uP stat/nM t/T`
+, the stored entry will be `John Doe Phone: 98765432 Email: johnd@example.com Address: NUS #04-02 Tags: [Tutor][Upper Pri][50][Physics][Not Matched]`
+
+=== Filter feature
+
+==== Current Implementation
+We have added the `filter` command which inherit from `Command`. The user can `filter` according to the attributes `Subject`, `Level` and `Status` of a person.
+
+The following sequence diagram shows how the filter command works:
+
+image::FilterCommandSequenceDiagram.png[width="800"]
+
+[NOTE]
+The user must input arguments in case sensitive and full fashion.
+For instance, the entry `filter maTH` will not work.
+
+=== UI Modifications
+
+==== Browser Panel
+The browser panel is modified to show the selected person’s details instead of opening up a URL.
+This allows better visualization of a person’s details as each person now has up 9 attributes: `Name`, `Phone`,
+`Address`, `Email`, `Role`, `Status`, `Subject`, `Level`, `Price` to be displayed.
+Below shows the modifications made:
+
+**Before**
+
+image::OldBrowserPanel.png[width="500"]
+
+**After**
+
+image::NewBrowserPanel.png[width="500"]
```

Comment:
You can show the composition of the new browser panel using a class diagram, if applicable.